Source/macOS/OIDRedirectHTTPHandler.m

```diff
@@ -76,10 +76,18 @@ - (NSURL *)startHTTPListener:(NSError **)returnError {
       *returnError = error;
     }
     return nil;
-  } else {
+  } else if ([_httpServ hasIPv4Socket]) {
```

Comment:
The order of these clauses would cause the library to always use IPv4, since, with your changes, the library always must have an IPv4 socket, but only may have an IPv6 socket, this first test will always pass, leaving no chance for the second test.